state/mongodb/metadata.yaml

```diff
@@ -17,34 +17,49 @@ capabilities:
   - transactional
   - etag
   - query
-metadata:
-  - name: server
-    # Required if host is not set
-    required: false
-    description: |
-      The server to connect to, when using DNS SRV record. One of `server` and `host` is required.
-    example: '"server.example.com"'
-  - name: host
-    # Required if server is not set
-    required: false
-    description: |
-      The host to connect to. One of `server` and `host` is required.
-    example: '"mongo-mongodb.default.svc.cluster.local:27017"'
-  - name: connectionstring
-    # Required if host and server are not set
-    required: false
+authenticationProfiles:
+  - title: "Connection string based authentication"
     description: |
-      The connection string to use. One of `server` or `host` or `connectionstring` is required.
-    example: '"mongodb://localhost:27017"'
-  - name: username
+      Authenticate using a connetion string.
+    metadata:
+      - name: connectionString
+        required: true
+        description: |
+          The connection string to use. One of `server` or `host` or `connectionstring` is required.
```

Comment:
```suggestion
          The connection string to use. One of `server`, `host`, and `connectionstring` is required.
```

Reply:
changed to 
`The connection string to use. One of `server`, `host`, or `connectionstring` is required.`